lib/util/utils.js

```diff
@@ -580,39 +580,133 @@ exports.relaxModelLikeEntities = function relaxModelLikeEntities(model) {
   return model;
 }
 
+/**
+ * Relaxes the entity to be a oneOf: [the current type OR null type] if the condition is satisfied
+ * @param {object} entity - The entity to be relaxed
+ * @param {Boolean|undefined} isPropRequired - states whether the property is required.
+ * If true then it is required. If false or undefined then it is not required.
+ * @returns {object} entity - The processed entity
+ */
 exports.allowNullType = function allowNullType(entity, isPropRequired) {
-  if (entity.type) { //not sure if I need to consider additionalProperties
+  // if entity has a type
+  if (entity && entity.type) {
+    // if type is an array
+    if (entity.type === "array") {
+      if (entity.items) {
+        // if items object contains inline properties
+        if (entity.items.properties) {
+          entity.items = exports.allowNullableTypes(entity.items)
+        } else {
+          entity.items = exports.allowNullType(entity.items);
+        }
+      }
+    }
+
+    // takes care of string 'false' and 'true'
+    if (typeof entity['x-nullable'] === 'string') {
+      if (entity['x-nullable'].toLowerCase() === 'false') {
+        entity['x-nullable'] = false;
+      } else if (entity['x-nullable'].toLowerCase() === 'true') {
+        entity['x-nullable'] = true;
+      }
+    }
+
+    let isPropNullable = entity['x-nullable'] && typeof entity['x-nullable'] === 'boolean';
+    if ((isPropNullable === undefined && !isPropRequired) || isPropNullable) {
```

Comment:
It is worthwhile providing a comment or may be putting a True False logic table that we had drawn on the white board in the comments over here. That will be helpful to us in the future.